File: Cpp/852_Peak_Index_in_a_Mountain_Array.cpp
class Solution {
public:
    int peakIndexInMountainArray(vector<int>& A) {
        int low = 0;
        int high = A.size()-1;
        
        while(low<=high){
            int mid = (low + high) / 2;
            if(A[mid] > A[mid-1] && A[mid] > A[mid+1]) return mid;
            if(A[mid] < A[mid+1]) low = mid + 1;
            else high = mid - 1;
        }
        
        return 0;
    }
};

File: Kotlin/852_Peak_Index_in_a_Mountain_Array.kt
class Solution {
    fun peakIndexInMountainArray(arr: IntArray): Int {
        var lo = 0
        var hi = arr.lastIndex
        
        while(lo < hi){
            val mid = (lo + hi) / 2
            if(arr[mid] < arr[mid + 1]) lo = mid + 1
            else hi = mid
        }
        
        return lo
    }
}